packages/schema/src/model.ts

```diff
@@ -16,25 +16,69 @@ import {
   IJupyterCadModel,
   ISelection,
   IUserData,
-  Pointer
+  Pointer,
+  IJCadSettings
 } from './interfaces';
 import jcadSchema from './schema/jcad.json';
 import { Contents } from '@jupyterlab/services';
+import { ISettingRegistry } from '@jupyterlab/settingregistry';
+
+const SETTINGS_ID = '@jupytercad/jupytercad-core:jupytercad-settings';
 
 export class JupyterCadModel implements IJupyterCadModel {
   constructor(options: JupyterCadModel.IOptions) {
-    const { annotationModel, sharedModel } = options;
+    const { annotationModel, sharedModel, settingRegistry } = options;
     if (sharedModel) {
       this._sharedModel = sharedModel;
     } else {
       this._sharedModel = this.createSharedModel();
     }
     this._connectSignal();
     this.annotationModel = annotationModel;
+    this.settingRegistry = settingRegistry;
     this._copiedObject = null;
     this._pathChanged = new Signal<JupyterCadModel, string>(this);
   }
 
+  /**
+   * Initialize custom settings for JupyterLab.
+   */
+  readonly settingsChanged = new Signal<this, void>(this);
+
+  async initSettings() {
+    if (this.settingRegistry) {
+      const setting = await this.settingRegistry.load(SETTINGS_ID);
+      this._settings = setting;
+
+      setting.changed.connect(() => {
+        this.settingsChanged.emit();
```

Comment:
you should emit the changed value so that the main view knows which thing need to be updated

Reply:
Done in my custom signal